junit5/src/main/java/cz/xtf/junit5/extensions/OpenShiftRecorderHandler.java

```diff
@@ -57,20 +60,143 @@
  */
-public class OpenShiftRecorderHandler
-        implements TestWatcher, TestExecutionExceptionHandler, BeforeAllCallback, LifecycleMethodExecutionExceptionHandler {
+public class OpenShiftRecorderHandler implements TestWatcher, TestExecutionExceptionHandler, BeforeAllCallback,
+        BeforeEachCallback, LifecycleMethodExecutionExceptionHandler {
     private static final Logger log = LoggerFactory.getLogger(OpenShiftRecorderHandler.class);
 
-    private static final String TRACK_FROM = "TRACK_FROM";
+    private static final String FILTER_FOR_BEFORE_ALL_DONE = "RESOURCES_FOR_BEFORE_ALL_RECORDED";
+
+    private static final String POD_FILTER_MASTER = "POD_FILTER_MASTER";
+    private static final String DC_FILTER_MASTER = "DC_FILTER_MASTER";
+    private static final String BUILD_FILTER_MASTER = "BUILD_FILTER_MASTER";
+    private static final String BC_FILTER_MASTER = "BC_FILTER_MASTER";
+    private static final String IS_FILTER_MASTER = "IS_FILTER_MASTER";
+    private static final String SS_FILTER_MASTER = "SS_FILTER_MASTER";
+    private static final String ROUTE_FILTER_MASTER = "ROUTE_FILTER_MASTER";
+    private static final String SERVICE_FILTER_MASTER = "SERVICE_FILTER_MASTER";
+    private static final String EVENT_FILTER_MASTER = "EVENT_FILTER_MASTER";
+
+    private static final String POD_FILTER_BUILDS = "POD_FILTER_BUILDS";
+    private static final String BUILD_FILTER_BUILDS = "BUILD_METHOD_FILTER_BUILDS";
+    private static final String BC_FILTER_BUILDS = "BC_FILTER_BUILDS";
+    private static final String IS_FILTER_BUILDS = "IS_FILTER_BUILDS";
+    private static final String EVENT_FILTER_BUILDS = "EVENT_FILTER_BUILDS";
 
     @Override
-    public void beforeAll(ExtensionContext extensionContext) throws Exception {
-        getClassStore(extensionContext).put(TRACK_FROM, EventHelper.timeOfLastEventBMOrTestNamespaceOrEpoch());
+    public void beforeAll(ExtensionContext context) {
+        Store classStore = getClassStore(context);
+        OpenShift master = OpenShifts.master();
+        OpenShift bm = BuildManagers.get().openShift();
+
+        initClassFilter(context, POD_FILTER_MASTER, master, Pod.class);
+        initClassFilter(context, DC_FILTER_MASTER, master, DeploymentConfig.class);
+        initClassFilter(context, BUILD_FILTER_MASTER, master, Build.class);
+        initClassFilter(context, BC_FILTER_MASTER, master, BuildConfig.class);
+        initClassFilter(context, IS_FILTER_MASTER, master, ImageStream.class);
+        initClassFilter(context, SS_FILTER_MASTER, master, StatefulSet.class);
+        initClassFilter(context, ROUTE_FILTER_MASTER, master, Route.class);
+        initClassFilter(context, SERVICE_FILTER_MASTER, master, Service.class);
+        classStore.put(EVENT_FILTER_MASTER,
+                new EventsFilterBuilder().setExcludedUntil(ResourcesTimestampHelper.timeOfLastEvent(master)));
+
+        initClassFilter(context, POD_FILTER_BUILDS, bm, Pod.class);
```

Comment:
I think there should be handled case when `build namespace == test.namespace`. In this case build namespace would not be recorded. This will need changes in other parts of the code as well.

Reply:
Yeah, good catch.

Reply:
done